securedrop_client/gui/conversation/print/printer.py

```diff
@@ -0,0 +1,125 @@
+from typing import NewType
+
+from PyQt5.QtCore import QObject, QTimer, pyqtSignal, pyqtSlot
+
+from securedrop_client import export
+from securedrop_client.logic import Controller
+
+
+class Printer(QObject):
+
+    status_changed = pyqtSignal()
+    job_done = pyqtSignal(str)
+    job_failed = pyqtSignal(str, str)
+
+    Status = NewType("Status", str)
+    StatusUnknown = Status("unknown-sf5fd")
+    StatusReady = Status("ready-83jf3")
+    StatusUnreachable = Status("unreachable-120a0")
+    # TODO StatusBusy = Status("busy-4bf83")
+
+    def __init__(self, controller: Controller, export_service: export.Service) -> None:
+        super().__init__()
+
+        self._status = self.StatusUnknown
+
+        self._printing_service = self._ExportServiceConnector(self, export_service)
+
+    @property
+    def status(self) -> Status:
+        return self._status
+
+    @status.setter
+    def status(self, new: Status) -> None:
+        if new != self._status:
+            self._status = new
+            self.status_changed.emit()
+
+    def start_and_watch_on(self, signal: pyqtSignal) -> None:
+        """Allows print queue to be started and its status to be watched regularly, in a thread-safe manner."""  # noqa: E501
+        signal.connect(self._printing_service._start_and_watch)
+
+    def stop_watching_on(self, signal: pyqtSignal) -> None:
+        """Allows to stop watching the print queue status in a thread-safe manner."""
+        signal.connect(self._printing_service._stop_watching)
```

Comment:
These two methods, along with the `Printer._ExportServiceConnector._DEFAULT_POLLING_DELAY_IN_MILLISECONDS` constitute the entire polling API. :slightly_smiling_face: 

The consuming component is responsible for making sure that it is registering as many starting signals as desired (starting and watching the print queue is an idempotent operation in practice), and matching that with however may signals are emitted when the printing operations finish.